phases/account.go

```diff
@@ -1,9 +1,106 @@
 package phases
 
-import "fmt"
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"net/http"
 
-// Account ...
-func Account() (string, error) {
-	fmt.Println("SetAccount")
-	return "", nil
+	"github.com/bitrise-io/go-utils/log"
+)
+
+type organizationData struct{
+	Name string
+	Slug string
+}
+
+func isValid(choice int, limit int ) bool {
+	return choice >= 1 && choice <= limit
+}
+
+// Account returns the slug of the selected account. If the user selects
+// the personal account, the slug is empty.
+func Account(apiToken string) (string, error) {
+	req, err := http.NewRequest(http.MethodGet, "https://api.bitrise.io/v0.1/organizations", nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Set("Authorization", "token " + apiToken)
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if res.StatusCode != 200 {
+		return "", fmt.Errorf("fetch orgs: %s", res.Status)
+	}
+
+	m := struct{
+		Data []organizationData
+	}{}
+	if err := json.NewDecoder(res.Body).Decode(&m); err != nil {
+		return "", err
+	}
+	
+	req, err = http.NewRequest(http.MethodGet, "https://api.bitrise.io/v0.1/me", nil)
+	if err != nil {
+		return "", err
+	}
+	
+	req.Header.Set("Authorization", "token " + apiToken)
+	res, err = http.DefaultClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+	
+	if res.StatusCode != 200 {
+		return "", fmt.Errorf("fetch user: %s", res.Status)
+	}
+	
+	u := struct{
+		Data struct{
+			Username string
+		}
+	}{}
+	if err := json.NewDecoder(res.Body).Decode(&u); err != nil {
+		return "", err
+	}
+	
+	options := []organizationData{organizationData{Name: u.Data.Username}}
+	options = append(options, m.Data...)
+	
+	log.Infof("ACCOUNT OPTIONS")
+	log.Infof("===============")
+	for i, opt := range options {
+		log.Printf("%d) %s", i + 1, opt.Name)
+	}
+
+	var choice int
+	for !isValid(choice, len(options)) {
+		log.Warnf("CHOOSE ACCOUNT: ")
+		
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			log.Warnf("error reading choice from stdin: %s", err)
+			continue
+		}
+		
+		choice, err = strconv.Atoi(strings.TrimSpace(input))
+		if err != nil {
+			log.Warnf("error reading choice from stdin: %s", err)
+			continue
+		} else if !isValid(choice, len(options)) {
+			log.Warnf("invalid choice")
+			continue
+		} else {
+			break
```

Comment:
you can have the check(`isValid(choice, len(options))`) here so you can skip couple of method calls in case it is not neccessary